src/zope/configuration/config.py

```diff
@@ -37,7 +37,36 @@
 from zope.configuration._compat import string_types
 from zope.configuration._compat import text_type
 
-
+__all__ = [
+    'ConfigurationContext',
+    'ConfigurationAdapterRegistry',
+    'ConfigurationMachine',
+    'ConfigurationExecutionError',
+    'IStackItem',
+    'SimpleStackItem',
+    'RootStackItem',
+    'GroupingStackItem',
+    'ComplexStackItem',
+    'GroupingContextDecorator',
+    'DirectiveSchema',
+    'IDirectivesInfo',
+    'IDirectivesContext',
+    'DirectivesHandler',
+    'IDirectiveInfo',
+    'IFullInfo',
+    'IStandaloneDirectiveInfo',
+    'defineSimpleDirective',
+    'defineGroupingDirective',
+    'IComplexDirectiveContext',
+    'ComplexDirectiveDefinition',
+    'subdirective',
+    'IProvidesDirectiveInfo',
+    'provides',
+    'toargs',
+    'expand_action',
+    'resolveConflicts',
+    'ConfigurationConflictError',
+]
```

Comment:
Maybe you could sort the lines in the list. (Here and in the other modules.)

Reply:
I put them in the order they are documented in the .rst files. There does (sometimes) seem to be some sort of logical grouping. For instance, `IInclude` is for the three directives `include`, `exclude` and `includeOverrides` and they are in that order in documentation. When we switch to `.. automodule::` it follows the order of `__all__` by default IIRC.